docs/DeveloperGuide.md

```diff
@@ -6,24 +6,177 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Parser component
+The Parser is essential to interpreting user commands and executing appropriate actions on the timetable. 
+This includes command parsing; breaking down user inputs into commands and parameters that PlaNus can understand and
+act upon.
 
+Parser interacts with the following classes.
+- 'Course': Creates Courses based on user input
+- 'Timetable': Modifies timetable based on user input
+- 'Storage': Uses Storage to save user data
+- 'UI': Uses Ui to display error messages
+
+Below is the class diagram:
+
+![Parser_ClassDiagram](uml/Parser_ClassDiagram.png)
+
+<br>
+
+### Timetable plan component
+In order to keep track of the timetable plan of the user, 3 classes are used mainly.
+- `Course`: Represents a course taken or planned by the user
+- `Timetable`: Stores the list of courses
+- `PlanGetter`: Provides access to the timetable plan
+
+Below is the class diagram to illustrate this:
+
+![TimetablePlan_ClassDiagram](uml/TimetablePlan_ClassDiagram.png)
+
+<br>
+
+When `addCourse()` is called, it finds the suitable position to insert the newly added course according to the 
+year and term. Then, the new course will be added into the list of courses.
+
+Below is the sequence diagram to illustrate the implementation:
+
+![AddCourse_SequenceDiagram](uml/AddCourse_SequenceDiagram.png)
+
+<br>
+
+When `removeCourse()` is called, it attempts to find the course with the given course code and removes the course 
+from the list of courses if it exists.
+
+Below is the sequence diagram to illustrate the implementation:
+
+![RemoveCourse_SequenceDiagram](uml/RemoveCourse_SequenceDiagram.png)
+
+<br>
+
+There are 3 overloaded variations of the `getPlan()` method in the `PlanGetter` class.
```

Comment:
Would it have been better to introduce these overloaded methods in the above diagram as well? Otherwise, since it is referencing "the above diagram", it may be better to position this section right under the class diagram for ease of reference as well.